### PR TITLE
Assignment 2 - done

### PR DESCRIPTION
**Pull request by Dmytro-Bonislavskyi**
## Checklist
- [1 ] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by ananyajha2000 — COMMENTED** [review]
Incorrect handle unknown for OneHotEncoder. Please fix and resubmit

**Comment by Dmytro-Bonislavskyi** [comment]
> Incorrect handle unknown for OneHotEncoder. Please fix and resubmit

Hello @ananyajha2000, could you, please, specify the issue.
In the task provided, we have to: Apply sklearn.preprocessing.OneHotEncoder:
        + Handle unknown labels if they exist.  -> `handle_unknown='ignore'`
        + Drop one column for binary variables. -> `drop='if_binary'`

If you mean warning that appear on cross validation stage: "UserWarning: Found unknown categories in columns [7] during transform. These unknown categories will be encoded as all zeros"

This warning appears because X_Train set (with random_state = 42) include one additional category in 'native-country' -> " Holand-Netherlands" which is not presented in X_Test set. 
After reading "transform" function from https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/preprocessing/_encoders.py, I can consider that this behaviour is intended. This warning is informative and ensures transparency in how the transformer handles new, unseen categories during the transformation process.

Maybe I missed something.

**Comment by ananyajha2000** [comment]
cat_transforms = Pipeline([
    ('impute', SimpleImputer(strategy='most_frequent')),
    ('onehot', OneHotEncoder(handle_unknown='infrequent_if_exist', drop='if_binary')),      
])

^This is what you need to update

**Comment by Dmytro-Bonislavskyi** [comment]
Hello, @ananyajha2000!
TYhank you for your reply. I changed the handle_unknown strategy for OneHotEncoder from 'ignore' to 'infrequent_if_exist' as you requested.

I know that Jesus used 'infrequent_if_exist' in his example, but since the current task did not specify which strategy we have to use, I chose another one. Anyway, now it's changed as you suggested.

**Comment by ananyajha2000** [comment]
From the assignment description:
-Apply [one-hot encoding](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html):
   - **Handle unknown labels if they exist.**
   - Drop one column for binary variables.

**Comment by ananyajha2000** [comment]
Marked as completed
